### PR TITLE
Refactored KuCoin deserialization methods and addressed tests

### PR DESCRIPTION
**Pull request by vnavascues**
Original issue #297
Related #2336 #2371 #2373 

Pending to address the Sandbox tests that fail due to invalid API keys. KuCoin sandbox is down at the moment.

### CODE REVIEW

File: rotkehlchen/tests/exchanges/test_kucoin.py [hunk 1]
```diff
@@ -753,14 +736,22 @@ def mock_api_query_response(case, options):  # pylint: disable=unused-argument
         return MockResponse(HTTPStatus.OK, next(get_response))
 
     get_response = get_endpoints_response()
-    with patch.object(
+    # Force a times_stap of 2s
```
Comment (LefterisJP) [inline rotkehlchen/tests/exchanges/test_kucoin.py:739]: times_stap -> time_step

File: rotkehlchen/tests/exchanges/test_kucoin.py [hunk 2]
```diff
@@ -727,24 +695,39 @@ def test_query_asset_movements_sandbox(
...
 
     def get_endpoints_response():
         results = [
-            f'{deposits_response}',
-            f'{withdrawals_response}',
+            f'{deposits_response_1}',
+            f'{deposits_response_2}',
+            f'{withdrawals_response_1}',
+            f'{withdrawals_response_2}',
+            f'{withdrawals_response_3}',  # never requested as
```
Comment (LefterisJP) [inline rotkehlchen/tests/exchanges/test_kucoin.py:730]: what do you mean never requested as?

**Review by LefterisJP — APPROVED** [review]
Seems to work with 1 account I tested. Left some small comments.

Can you also remove something I forgot? I left a print statement in kucoin.py:376 :sweat:

File: rotkehlchen/tests/exchanges/test_kucoin.py (hunk 2 above)
Comment (vnavascues) [reply rotkehlchen/tests/exchanges/test_kucoin.py:730]: The `as` is a typo, "is never requested" or I can just remove the comment.
If pagination works as expected (breaking the requesting loop), this withdrawal won't be processed.

Comment (LefterisJP) [reply rotkehlchen/tests/exchanges/test_kucoin.py:730]: perhaps just modify the comment explaining what you just said here :)

**Comment by vnavascues** [comment]
> Seems to work with 1 account I tested. Left some small comments.
> 
> Can you also remove something I forgot? I left a print statement in kucoin.py:376

I've applied the small changes. Fingers crossed, and many thanks for addressing the pagination.

**Review by LefterisJP — APPROVED** [review]
lgtm thanks

**Pull request merged**